pkg/core/fs_rw_ops.go

```diff
@@ -548,6 +547,170 @@ func (fs *fsMutable) ReleaseFileHandle(
 	return
 }
 
+// Commit changes on a MutableFS
+func (dfs *MutableFS) Commit() error {
+	return dfs.fsInternal.Commit()
+}
+
+// Add the root of the FS into the FS.
+func (fs *fsMutable) initRoot() (err error) {
+	_, found := fs.lookupTree.Get(formKey(fuseops.RootInodeID))
+	if found {
+		return
+	}
+	err = fs.createNode(
+		formLookupKey(fuseops.RootInodeID, rootPath),
+		fuseops.RootInodeID,
+		rootPath,
+		nil,
+		fuseutil.DT_Directory,
+		true)
+	return
+}
+
+// Run validations before creating a node. Need to take locks before calling.
+func (fs *fsMutable) preCreateCheck(parentInode fuseops.InodeID, lk []byte) error {
+	// check parent exists
+	key := formKey(parentInode)
+	e, found := fs.iNodeStore.Get(key)
+	if !found {
+		return fuse.ENOENT
+	}
+
+	// parent is a directory
+	n := e.(*nodeEntry)
+	if !n.attr.Mode.IsDir() {
+		return fuse.ENOTDIR
+	}
+
+	// check child name not taken
+	_, found = fs.lookupTree.Get(lk)
+	if found {
+		return fuse.EEXIST
+	}
+	return nil
+}
+
+func (fs *fsMutable) insertReadDirEntry(id fuseops.InodeID, dirEnt *fuseutil.Dirent) {
+
+	if fs.readDirMap[id] == nil {
+		fs.readDirMap[id] = make(map[fuseops.InodeID]*fuseutil.Dirent)
+	}
+	fs.readDirMap[id][dirEnt.Inode] = dirEnt
+}
+
+func (fs *fsMutable) insertLookupEntry(id fuseops.InodeID, child string, entry lookupEntry) {
+	fs.lookupTree, _, _ = fs.lookupTree.Insert(formLookupKey(id, child), entry)
+}
+
+// Create a node. Need to hold the locks before calling.
+func (fs *fsMutable) createNode(lk []byte, parentINode fuseops.InodeID, childName string,
```

Comment:
another case where we're shuffling physical layout (move `createNode` here from `fs_common.go`) ... and possibly changing logic as well?  i cannot easily tell what changed b/c git is recording it as a rewrite.

Reply:
yes that is unfortunate, but that's the way it goes when a repo has accumulated too much debt

Reply:
follow-up note: open discussion thread

Reply:
I am open to dropping this functionality for now.

Reply:
Agree